src/SelfServe/SelfServeComponent.tsx

```diff
@@ -215,29 +259,62 @@ export class SelfServeComponent extends React.Component<SelfServeComponentProps,
     }
   };
 
-  public onSaveButtonClick = (): void => {
-    const onSavePromise = this.props.descriptor.onSave(this.state.currentValues);
-    onSavePromise.catch((error) => {
-      this.setState({
-        notification: {
-          message: `${error.message}`,
-          type: SelfServeNotificationType.error,
-        },
-      });
-    });
-    onSavePromise.then((notification: SelfServeNotification) => {
+  public performSave = async (): Promise<void> => {
+    this.setState({ isSaving: true, notification: undefined });
+    try {
+      const onSaveResult = await this.props.descriptor.onSave(
+        this.state.currentValues,
+        this.state.baselineValues as ReadonlyMap<string, SmartUiInput>
+      );
+      if (onSaveResult.portalNotification) {
+        const requestInitializedPortalNotification = onSaveResult.portalNotification.initialize;
+        const requestSucceededPortalNotification = onSaveResult.portalNotification.success;
+        const requestFailedPortalNotification = onSaveResult.portalNotification.failure;
+
+        this.sendNotificationMessage({
+          retryIntervalInMs: this.retryIntervalInMs,
+          operationStatusUrl: onSaveResult.operationStatusUrl,
+          portalNotification: {
+            initialize: {
+              title: this.getTranslation(requestInitializedPortalNotification.titleTKey),
+              message: this.getTranslation(requestInitializedPortalNotification.messageTKey),
+            },
+            success: {
+              title: this.getTranslation(requestSucceededPortalNotification.titleTKey),
+              message: this.getTranslation(requestSucceededPortalNotification.messageTKey),
+            },
+            failure: {
+              title: this.getTranslation(requestFailedPortalNotification.titleTKey),
+              message: this.getTranslation(requestFailedPortalNotification.messageTKey),
+            },
+          },
+        });
+      }
+      promiseRetry(() => this.pollRefresh(), this.retryOptions);
+    } catch (error) {
       this.setState({
         notification: {
-          message: notification.message,
-          type: notification.type,
+          type: MessageBarType.error,
+          isCancellable: true,
+          message: this.getTranslation(error.message),
         },
       });
-      this.resetBaselineValues();
-      this.onRefreshClicked();
-    });
+      throw error;
```

Comment:
Do we need to log the error?

Reply:
the error is set in the notification bar